app/models/activity.rb

```diff
@@ -115,6 +115,11 @@ def student_only
       subject_area_list
     end
     integer :project_ids, :multiple => true, :references => Admin::Project
+
+    string :project_names, :multiple => true do
+      projects.map { |p| p.name }
+    end
+
```

Comment:
I don't think this is necessary unless I'm missing something.  The advanced search, searches on project_id not project name. And I don't see anything in this PR or in the PDF mockups that requires changes to searching by project.  (ditto for the other material types)